### PR TITLE
Migrate examples to the Android embedding v2

### LINKED ISSUE #33884: Convert Flutter framework repo to new Android embedding
- [ ] devicelab tests
- [ ] examples
- [x] dev/integration_test

### PR DESCRIPTION
**Pull request by blasten**
## Description

Migrate the examples apps to the Android embedding v2.

## Related Issues

Fixes https://github.com/flutter/flutter/issues/33884

Related to https://github.com/flutter/flutter/issues/43220

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Review by matthew-carroll — COMMENTED** [review]
LGTM

**Review by matthew-carroll — APPROVED** [review]

File: dev/benchmarks/macrobenchmarks/android/app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -35,5 +35,10 @@
                 <category android:name="android.intent.category.LAUNCHER"/>
             </intent-filter>
         </activity>
+    <!-- Don't delete the meta-data below.
```
Comment (xster) [inline dev/benchmarks/macrobenchmarks/android/app/src/main/AndroidManifest.xml]: Was this what we decided as opposed to a string search for import "io.flutter.embedding" in the users' java code?

**Review by xster — COMMENTED** [review]
LGTM

File: dev/benchmarks/macrobenchmarks/android/app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (blasten) [reply dev/benchmarks/macrobenchmarks/android/app/src/main/AndroidManifest.xml]: yeah. wdyt?

Comment (xster) [reply dev/benchmarks/macrobenchmarks/android/app/src/main/AndroidManifest.xml]: Sure, SGTM. What happens when there's mismatches? (false positives or negatives)

Comment (blasten) [reply dev/benchmarks/macrobenchmarks/android/app/src/main/AndroidManifest.xml]: I’m hoping we can automize the migration, so apps devs don’t have to modify the manifest manually. But yes, if someone changes this manually without also changing the java code , then the app won’t compile.

Comment (xster) [reply dev/benchmarks/macrobenchmarks/android/app/src/main/AndroidManifest.xml]: Ok, SG

**Pull request merged**